nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-restapi-processors/src/test/java/org/apache/nifi/processors/elasticsearch/PutElasticsearchRecordTest.java

```diff
@@ -0,0 +1,784 @@
+import org.apache.nifi.json.JsonRecordSetWriter;
+import org.apache.nifi.json.JsonTreeReader;
+import org.apache.nifi.processors.elasticsearch.mock.MockBulkLoadClientService;
+import org.apache.nifi.provenance.ProvenanceEventType;
+import org.apache.nifi.schema.access.SchemaAccessUtils;
+import org.apache.nifi.serialization.RecordReaderFactory;
+import org.apache.nifi.serialization.record.MockRecordParser;
+import org.apache.nifi.serialization.record.MockSchemaRegistry;
+import org.apache.nifi.serialization.record.RecordFieldType;
+import org.apache.nifi.util.MockFlowFile;
+import org.apache.nifi.util.StringUtils;
+import org.apache.nifi.util.TestRunner;
+import org.apache.nifi.util.TestRunners;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TimeZone;
+import java.util.function.Consumer;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class PutElasticsearchRecordTest extends AbstractPutElasticsearchTest<PutElasticsearchRecord> {
+    private static final int DATE_YEAR = 2020;
+    private static final int DATE_MONTH = 11;
+    private static final int DATE_DAY = 27;
+    private static final int TIME_HOUR = 12;
+    private static final int TIME_MINUTE = 55;
+    private static final int TIME_SECOND = 23;
+    private static final LocalDateTime LOCAL_DATE_TIME = LocalDateTime.of(DATE_YEAR, DATE_MONTH, DATE_DAY, TIME_HOUR, TIME_MINUTE, TIME_SECOND);
+    private static final LocalDate LOCAL_DATE = LocalDate.of(DATE_YEAR, DATE_MONTH, DATE_DAY);
+    private static final LocalTime LOCAL_TIME = LocalTime.of(TIME_HOUR, TIME_MINUTE, TIME_SECOND);
+    private static final String TEST_DIR = "src/test/resources/PutElasticsearchRecordTest";
+    private static final String TEST_COMMON_DIR = "src/test/resources/common";
+    private static String simpleSchema;
+    private MockBulkLoadClientService clientService;
+    private MockSchemaRegistry registry;
+    private TestRunner runner;
+
+    @Override
+    public Class<? extends AbstractPutElasticsearch> getTestProcessor() {
+        return PutElasticsearchRecord.class;
+    }
+
+    @BeforeAll
+    public static void setUpBeforeClass() throws Exception {
+        simpleSchema = Files.readString(Paths.get(TEST_DIR, "simpleSchema.json"));
+        TimeZone.setDefault(TimeZone.getTimeZone("UTC"));
```

Comment:
Needs a closer look - this doesn't seem right

Reply:
@exceptionfactory I do not know why this did not surface earlier. I did not make any changes to assertion code. I cut and pasted it from the previous Groovy closure to a Java Consumer.

Reply:
I wonder whether this is due to the Groovy tests having converted, for example, a `LocalDateTime` into a `Timestamp`, to an `Instant`, and finally the `long` representation of the "epoch millis" - this would (I think) cater for the Locale/TimeZone change in the JVM, e.g. a `Europe/Paris` TZ instead of `UTC`, allowing the tests to use a consistent basis for creating, using, and then asserting the timestamp/data/time

With the change to a JSON-based structure for the FlowFile contents, are the timestamps being read from the file in one TZ then the tests and trying to use/assert them in another?

I recall having to spend some time (no pun intended) on getting these tests to work originally, but it was a while ago now, so can't recall the specifics (think I maybe tried to start with a hardcoded timestamp-millis, but couldn't figure out how to get that to work)

Reply:
@ChrisSamo632 @exceptionfactory This information helped a lot. I tweeked my code to modify the read in JSON files with the same values you had used with 
`Time.valueOf(LOCAL_TIME).getTime()`
`Timestamp.valueOf(LOCAL_DATE_TIME).toInstant().toEpochMilli())`
`Date.valueOf(LOCAL_DATE).getTime())`

Now even with changing the timezone with
`TimeZone.setDefault(TimeZone.getTimeZone("Europe/Paris"));`
the tests pass.